File: frontend/src/models/Poll.ts
export class ProposedDate {
    date: Date;
    duration: string;
    voterIds: string[];

    constructor(date: Date, duration: string, voterIds: string[]) {
        this.date = date;
        this.duration = duration;
        this.voterIds = voterIds;
    }
}

export class Poll {
    id?: string;
    organizerId: string;
    title: string;
    description: string;
    location: string;
    participantIds: string[];
    proposedDates: ProposedDate[];
    bookedDateIndex?: number;

    constructor(id: string, organizerId: string, title: string, description: string, location: string, participantsIds: string[], proposedDates: ProposedDate[], bookedDateIndex: number) {
        this.id = id;
        this.organizerId = organizerId;
        this.title = title;
        this.description = description;
        this.location = location;
        this.proposedDates = proposedDates;
        this.participantIds = participantsIds;
        this.bookedDateIndex = bookedDateIndex;
    }
}

File: backend/src/main/java/com/stroodle/backend/model/Poll.java
package com.stroodle.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Poll {
    @Id
    private String id;

    private String organizerId;

    private String title;

    private String description;

    private String location;

    private String duration;

    private List<String> participantIds;

    private List<ProposedDate> proposedDates;

    private Number bookedDateIndex;
}